File: ios/cli/UseSslPinningModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(UseSslPinning, NSObject)

RCT_EXTERN_METHOD(setUseSSLPinning:(BOOL)usePinning
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(getUseSSLPinning:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

// Removed initializeSslPinning method for consistency with workflow
// SSL pinning is auto-initialized in init() method

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/cli/UseSslPinningModule.swift
import Foundation
import React

/**
 * React Native CLI module for SSL Pinning
 * Uses the shared logic for common functionality
 */
@objc(UseSslPinning)
class UseSslPinning: NSObject {
    
    override init() {
        super.init()
        
        // Early initialization - setup SSL pinning if enabled
        let isEnabled = SharedLogic.getUseSSLPinning()
        if isEnabled {
            do {
                let _ = try SharedLogic.initializeSslPinningFromBundle()

            } catch {

            }
        } else {

        }
    }
    
    @objc
    func setUseSSLPinning(_ usePinning: Bool, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        SharedLogic.setUseSSLPinning(usePinning)
        resolve(nil)
    }
    
    @objc
    func getUseSSLPinning(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        let usePinning = SharedLogic.getUseSSLPinning()
        resolve(usePinning)
    }
    
    
}